### PR TITLE
Download agreements e2e tests

### PR DESCRIPTION
**Pull request by drakkein**

### CODE REVIEW

File: app/modules/immutable-file/sagas.ts [hunk 1]
```diff
@@ -17,6 +19,12 @@ export function* downloadFile(
     const immutableFileId = action.payload.immutableFileId;
     const downloadedFile = yield apiImmutableStorage.getFile(immutableFileId);
     const extension = immutableFileId.asPdf ? ".pdf" : ".doc";
+
+    if (IS_CYPRESS) {
+      // required for cypress to be able to simulate generating time
```
Comment (Moejoe90) [inline app/modules/immutable-file/sagas.ts]: Interesting trick how did you think of it?

File: app/components/portfolio/PortfolioMyAssets.tsx [hunk 2]
```diff
@@ -107,6 +107,8 @@ const PortfolioMyAssetsComponent: React.FunctionComponent<TComponentProps> = ({
                 iconPosition="icon-after"
                 svgIcon={arrowRight}
                 size={ButtonSize.SMALL}
+                data-test-id="portfolio-my-assets-neu-agreements"
+                data-test-wallet-address={walletAddress}
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioMyAssets.tsx]: is this used anywhere?
`data-test-wallet-address`

File: app/test-e2e/portfolio/PortfolioDownloadDocuments.spec.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          const downloadSelector = tid(
+            `modals.portfolio.portfolio-assets.download-agreements-${PUBLIC_ETO_ID}.download`,
+          );
+          cy.clock().then(clock => {
+            cy.get(downloadSelector)
+              .first()
+              .parent()
+              .click()
+              .then(() => {
+                cy.wait(100);
```
Comment (Moejoe90) [inline app/test-e2e/portfolio/PortfolioDownloadDocuments.spec.e2e.ts]: no need to for `wait` here

File: app/modules/immutable-file/sagas.ts (hunk 1 above)
Comment (drakkein) [reply app/modules/immutable-file/sagas.ts]: welll, `cy.clock` is taking control of every js time function includes `setTimeout` and `delay` is just `setTimeout` wrapped into `Promise`

File: app/components/portfolio/PortfolioMyAssets.tsx (hunk 2 above)
Comment (drakkein) [reply app/components/portfolio/PortfolioMyAssets.tsx]: no, it can be removed

**Pull request merged**